src/core/utils/KeyMapping.ts

```diff
@@ -0,0 +1,62 @@
+import { defaultKeyMap } from './DefaultKeyMap';
+
+export interface KeyModifiers {
+    ctrlKey?: boolean;
+    altKey?: boolean;
+    shiftKey?: boolean;
+}
+export interface Keypress {
+    key: string;
+    modifiers: KeyModifiers;
+}
+export interface IntentInformation {
+    name: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    arguments: Array<any> | Record<string, any>;
+}
+export type ShortcutIdentifier = string; // format: /[ctrl+][alt+][shift+]key/
+export type KeyMap = Map<ShortcutIdentifier, IntentInformation>;
+
+export class KeyMapping {
+    keyMap: KeyMap;
```

Comment:
Why the intermediate type ? `keyMap` is a `Map<ShortcutIdentifier, IntentInformation>`.

Reply:
Anticipating for custom keymaps, so you can import the type.